.pre-commit-hooks.yaml

```diff
@@ -206,6 +206,20 @@
       \.renovaterc\.(json|json5)
     )$
 
+# this hook is autogenerated from a script
+# to modify this hook, update `src/check_jsonschema/catalog.py`
+# and run `make generate-hooks` or `tox run -e generate-hooks-config`
+- id: check-snapcraft
+  name: Validate snapcraft files
+  description: 'Validate snapcraft files against the schema provided by SchemaStore'
+  entry: check-jsonschema --builtin-schema vendor.snapcraft
+  language: python
+  files: >
+    (?x)^(
+      ([^/]*/)*snapcraft.yaml
+    )$
```

Comment:
I need to read more to see where a `snapcraft.yaml` may appear, but my understanding is that it can go anywhere. So I think the desired pattern here is `snapcraft\.yaml` with no anchors. If the current hook generator doesn't support this, I think we should extend it rather than using a less nice pattern.

Reply:
I've made a small change here, but unfortunately the way this is tested doesn't let us have what I think is the simplest pattern. The test demands a full match, but I _think_ a suffix match would probably be best. I'll leave it only partly refined for now.